frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1569,6 +1643,59 @@ impl<T: Config> Pallet<T> {
 			})
 	}
 
+	/// Attempts to update the `MinimumUntrustedScore` if at least
+	/// `MinimumUntrustedScoreUpdateInterval` successful election have passed since the last
+	/// update. If not ready yet, update the minimum untrusted score rolling average in
+	/// the storage.
+	///
+	/// If the untrusted minimum score is updated, the new value is a percentage (defined by
+	/// `MinimumUntrustedScoreMargin`) of the rolling average of the previous election scores.
+	fn try_update_minimum_untrusted_score(score: ElectionScore) -> Result<(), &'static str> {
+		let update_interval = T::MinimumUntrustedScoreUpdateInterval::get().unwrap_or(0);
+		if update_interval == 0 {
+			// if update interval is not set, the on-chain update mechanism is disabled. return
+			// early.
+			return Ok(())
+		}
+
+		let (sliding_window_counter, rolling_average_score) =
+			<MinimumUntrustedScoreAvg<T>>::get().unwrap_or((0, ElectionScore::default()));
+
+		// update the rolling average score.
+		let score_weight = match score.checked_div(update_interval.into()) {
+			Some(d) => d,
+			// this error should not happen due to the check above, but since we are in
+			// `on_initialize` territory, we are extra careful and return early here. In practice
+			// the on-chain untrusted score update is disabled.
+			None => return Err("tried to divide score by zero, return early."),
+		};
+		let average_score = rolling_average_score.checked_add(&score_weight).unwrap_or(score);
```

Comment:
It would be nice to explain the calculation in comments. `average_score` is not really an average until we reach the update interval which was a bit unexpected for me given the name.

Reply:
It would be nice to explain the calculation in comments. `average_score` is not really an average until we reach the update interval which was a bit unexpected for me given the name.

Reply:
I added a check that if the interval value changes, the average is reset. I think this is the cleanest way to deal with this case, since we're not in a rush to update the minimum untrusted score and can easily cope with delays.

I added some comments on how the rolling average is calculated and renamed some vars.